### PR TITLE
added go mod tidy to the validate script

### PR DESCRIPTION
**Pull request by paynejacob**
Added go mod tidy to the validate script.  This will remove any unused dependencies before we build rancher.

https://github.com/rancher/rancher/issues/28515

### CODE REVIEW

**Review by rmweir — DISMISSED** [review]
LGTM

**Review by cbron — DISMISSED** [review]
lgtm. One thing is that this could turn the git status to dirty if it changes go.sum, which changes the tag in the version script, but I don't think that should actually cause any problems.

**Comment by dramich** [comment]
The build needs to fail if `go mod tidy` causes any changes

**Review by dramich — APPROVED** [review]

File: scripts/validate [hunk 1]
```diff
@@ -11,5 +11,13 @@ fi
 echo Running: golangci-lint
 golangci-lint run
 
+echo Tidying up modules
+go mod tidy
+
 echo Verifying modules
 go mod verify
+
+if [[ $(git diff --stat) != '' ]]; then
```
Comment (cbron) [inline scripts/validate:20]: Any reason we used `git diff` ? Typically we use [`git status --porcelain`](https://github.com/search?q=org%3Arancher+porcelain&type=code). Not a big difference in functionality, but could be parsing subtleties.

Comment (paynejacob) [reply scripts/validate:20]: I don't have a preference, ill switch it to porcelain to make it consistent.

**Review by cbron — APPROVED** [review]

**Pull request merged**